lib/superset/dashboard/import.rb

```diff
@@ -60,24 +67,56 @@ def route
         "dashboard/import/"
       end
 
-      def zip_database_config_not_found_in_superset
-        zip_databases_details.select {|s| !superset_database_uuids_found.include?(s[:uuid]) }
+      def zip?
+        File.extname(source) == ".zip"
       end
 
-      def superset_database_uuids_found
-        @superset_database_uuids_found ||= begin
-          zip_databases_details.map {|i| i[:uuid]}.map do |uuid|
-            uuid if Superset::Database::List.new(uuid_equals: uuid).result.present?
-          end.compact
+      def directory?
+        File.directory?(source)
+      end
+
+      def source_zip_file
+        return source if zip?
+
+        Zip::File.open(new_zip_file, Zip::File::CREATE) do |zipfile|
+          Dir[File.join(source, "**", "**")].each do |file|
+            zipfile.add(file.sub("#{source}/", "#{File.basename(source)}/"), file) if File.file?(file)
+          end
         end
+        new_zip_file
+      end
+
+      def new_zip_file
+        new_database_name = dashboard_config[:databases].first[:content][:database_name]
+        File.join(source, "dashboard_import.zip")
       end
```

Comment:
`new_database_name ` is not used and can be removed.

Can we add the timestamp to the newly create zip
and then reuse the timestamp in the base folder of the zip .. as superset looks to need a base folder
from what I have seen

```ruby
      def new_zip_file
        File.join(source, "dashboard_import_#{timestamp}.zip")
      end

      def timestamp
        @timestamp ||=Time.now.strftime("%Y%m%d%H%M%S")
      end
 ```